src/App.tsx

```diff
@@ -1,27 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import * as goodsFilterFunctionAPI from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const buttonClickHandle = async (loadedGoods: Promise<Good[]>) => (
+    setGoods(await loadedGoods)
+  );
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  return (
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    <div className="App">
+      <h1>Dynamic List of Goods</h1>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      <button
+        className="button"
+        type="button"
+        data-cy="all-button"
+        onClick={() => buttonClickHandle(goodsFilterFunctionAPI.getAll())}
```

Comment:
let's create a separate handler for each case, now it looks quite messy
```suggestion
        onClick={() => getAllTodos()}
```

Reply:
no, just get rid of `goodsFilterFunctionAPI` - it makes your code really complicated

Reply:
oh, ok, I see now

Reply:
this way you have it now makes much more sense